docs/circuit-breaking.md

```diff
@@ -0,0 +1,119 @@
+5. Allow monitoring of all Circuits, a "birds-eye" dashboard of the entire distributed system, and the ability to manually adjust Circuit state and parameters.
+
+### Terminology
+
+The jargon associated with Circuit Breaking makes the problem area more difficult to understand than it should be. To combat this, the following terms will be strictly used:
+
+| Term               | Definition                                                                          | Synonyms                      |
+| ------------------ | ----------------------------------------------------------------------------------- | ----------------------------- |
+| Caller             | A Service that is calling into another Service through a Circuit.                   | Source                        |
+| Circuit            | A unit that throttles traffic based on whether it is in Healthy or Unhealthy State. | Breaker, Bulkhead             |
+| Healthy            | A circuit state when traffic is flowing regularly.                                  | Closed Circuit, Reset Circuit |
+| Healthy Criteria   | A check that is used to determine when a Circuit should become Healthy.          |                               |
+| Probe              | A regular request that acts as a health check when a circuit is Unhealthy.          | Health Check                  |
+| Service            | An API/application that is using the Circuit Breaker pattern.                       | System, API, application      |
+| Unhealthy          | A circuit state when traffic is being denied, with occasional requests, called probes, being let through as health checks. | Open Circuit, Half-open Circuit |
+| Unhealthy Criteria | A check that is used to determine when a Circuit should become Unhealthy.        |                               |
+
+Phrases like "trip the circuit", "break the circuit", and "reset the circuit" are all explicitly avoided, instead opting for the more obvious "Healthy state change" and "Unhealthy state change".
+
+## Implementation
+
+The following diagram illustrates Hyperbahn's Circuit Breaking implementation:
+
+![Circuit Breaking](images/circuit_breaking.png)
+[Click Here For Full Size Diagram](images/circuit_breaking.png)
+
+
+* Each Service that is registered on the Hyperbahn maintains a set of Circuits that are responsible for denying traffic when the Circuit is Unhealthy.
+* The Circuits live on the Service's Egress nodes so that traffic can be denied before reaching the Service. The state of the Circuits can be optionally gossiped between that Service's Egress nodes to enhance the Circuit's sensitivity.
+* The Service maintains a Circuit for every Caller it has, times every endpoint it exposes, with a name format of `SOURCE->SERVICE::ENDPOINT`. In our diagram, this means that even though the `CatsForCharity->Petshop::listCats` Circuit is Unhealthy and denying traffic, the `PetBreeders->Petshop::listCats` Circuit is Healthy and continues to allow traffic to flow.
```

Comment:
That's an interesting point.

@Raynos @jcorbin has `arg1` in practice always always always been endpoint?